app/grandchallenge/algorithms/models.py

```diff
@@ -419,6 +431,21 @@ def create_result(self, *, result: dict):
             )
             self.outputs.add(output_civ)
 
+    @property
+    def rendered_result_text(self):
+        interface = ComponentInterface.objects.get(slug="results-json-file")
+        try:
+            output = self.outputs.get(interface=interface)
+            if not self.algorithm_image.algorithm.result_template:
+                return output.value
+            template_output = JINJA_ENGINE.from_string(
+                self.algorithm_image.algorithm.result_template
+            ).render(dict=output.value)
+
+            return md2html(template_output)
+        except (ObjectDoesNotExist, SecurityError):
```

Comment:
Could you split this into two try statements? That way, we know what exceptions we expect from what methods. So, one to get the output and catch `ObjectDoesNotExist`, and the second to catch `SecurityError` from the template rendering.

Reply:
Split the errors and added UndefinedError as well.